### PR TITLE
James-3080 use reworked rabbitMQrestart method in tests and make the code more resilient

### PR DESCRIPTION
**Pull request by remk**
follow up to #3163 with rebase on #3165

### CODE REVIEW

**Comment by Arsnael** [comment]
```
[5b7bf33ba7fa677abdd3717da427d5f133dfb2c2] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project apache-james-mailbox-event-rabbitmq: There was a timeout or other error in the fork -> [Help 1]
```
test this please

File: server/task/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQWorkQueue.java [hunk 1]
```diff
@@ -99,9 +104,16 @@ private void startWorkqueue() {
...
+        channelPool.getSender()
+            .declareExchange(ExchangeSpecification.exchange(EXCHANGE_NAME))
+            .retryBackoff(NUM_RETRIES, FIRST_BACKOFF, FOREVER)
+            .then(channelPool.getSender()
+                .declare(QueueSpecification.queue(QUEUE_NAME).durable(true).arguments(Constants.WITH_SINGLE_ACTIVE_CONSUMER))
+                .retryBackoff(NUM_RETRIES, FIRST_BACKOFF, FOREVER))
+            .then(channelPool.getSender()
+                .bind(BindingSpecification.binding(EXCHANGE_NAME, ROUTING_KEY, QUEUE_NAME))
+                .retryBackoff(NUM_RETRIES, FIRST_BACKOFF, FOREVER))
+            .block();
```
Comment (blackheaven) [inline server/task/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQWorkQueue.java]: I prefer that you split management operation (create a variable) and `then`s-chaining/`block`.

**Comment by nvduc91** [comment]
Read it

**Comment by remk** [comment]
```
[b8ad19030b15886f712a63cf0330862feef5cb44] [INFO] Tests run: 28, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 6.715 s - in org.apache.james.jmap.memory.cucumber.MemoryDownloadCucumberTest
[b8ad19030b15886f712a63cf0330862feef5cb44] [INFO] 
[b8ad19030b15886f712a63cf0330862feef5cb44] [INFO] Results:
[b8ad19030b15886f712a63cf0330862feef5cb44] [INFO] 
[b8ad19030b15886f712a63cf0330862feef5cb44] [ERROR] Errors: 
[b8ad19030b15886f712a63cf0330862feef5cb44] [ERROR]   MemorySpamAssassinContractTest.movingAMailToTrashShouldNotImpactSpamassassinLearning » ConditionTimeout
[b8ad19030b15886f712a63cf0330862feef5cb44] [ERROR]   MemorySpamAssassinContractTest.spamShouldBeDeliveredInSpamMailboxWhenSameMessageHasAlreadyBeenMovedToSpam » ConditionTimeout
```

test this please

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**